web/src/message_edit.js

```diff
@@ -701,17 +718,24 @@ export function toggle_resolve_topic(message_id, old_topic_name, report_errors_i
     channel.patch({
         url: "/json/messages/" + message_id,
         data: request,
+        success() {
+            const $spinner = $row.find(".toggle_resolve_topic_spinner");
+            loading.destroy_indicator($spinner);
+        },
         error(xhr) {
+            const $spinner = $row.find(".toggle_resolve_topic_spinner");
             if (xhr.responseJSON) {
                 if (xhr.responseJSON.code === "MOVE_MESSAGES_TIME_LIMIT_EXCEEDED") {
                     handle_resolve_topic_failure_due_to_time_limit(topic_is_resolved);
+                    loading.destroy_indicator($spinner);
```

Comment:
I think it might be better to just hide the spinner at the very start of the `error` block rather than calling it in both exit code paths.

Reply:
Makes sense. Done.